From eced9af4a77853fff3295391ccfa95faceece955 Mon Sep 17 00:00:00 2001
From: vdh-anssi <vincent.diemunsch@inria.fr>
Date: Fri, 24 Apr 2026 14:58:47 +0200
Subject: [PATCH 3/3] Bug fix of User Certificate ClientUserId

---
 src/server/ua_services_session.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/server/ua_services_session.c b/src/server/ua_services_session.c
index ec7d65b59..753f08d91 100644
--- a/src/server/ua_services_session.c
+++ b/src/server/ua_services_session.c
@@ -1219,8 +1219,8 @@ Service_ActivateSession_inner(UA_Server *server, UA_SecureChannel *channel,
     } else if(tokenType == &UA_TYPES[UA_TYPES_X509IDENTITYTOKEN]) {
         UA_X509IdentityToken* userCertToken = (UA_X509IdentityToken*)
             req->userIdentityToken.content.decoded.data;
-        UA_CertificateUtils_getSubjectName(&session->clientUserIdOfSession,
-                                           &userCertToken->certificateData);
+        UA_CertificateUtils_getSubjectName(&userCertToken->certificateData,
+                                           &session->clientUserIdOfSession);
     } else {
         /* TODO: Handle issued token */
     }
@@ -1314,7 +1314,7 @@ Service_CloseSession(UA_Server *server, UA_SecureChannel *channel,
     }
 #endif
 
-    /* Remove the sesison */
+    /* Remove the session */
     UA_Session_remove(server, session, UA_SHUTDOWNREASON_CLOSE);
 }
 
-- 
2.53.0

