From 7a3acfa7103f58f066805f678bf0d3ab26433f11 Mon Sep 17 00:00:00 2001
From: vdh-anssi <vincent.diemunsch@inria.fr>
Date: Fri, 9 Jan 2026 18:37:02 +0100
Subject: [PATCH 2/2] Added instrumentation to debug...

---
 src/server/ua_server.c | 31 +++++++++++++++++++++++++++++--
 1 file changed, 29 insertions(+), 2 deletions(-)

diff --git a/src/server/ua_server.c b/src/server/ua_server.c
index 979ba40e7..4a9f6eb25 100644
--- a/src/server/ua_server.c
+++ b/src/server/ua_server.c
@@ -466,7 +466,11 @@ UA_Server_delete(UA_Server *server) {
     /* Remove all server components (all stopped by now) */
     UA_ServerComponent *top;
     while((top = ZIP_ROOT(&server->serverComponents))) {
-        UA_assert(top->state == UA_LIFECYCLESTATE_STOPPED);
+        //UA_assert(top->state == UA_LIFECYCLESTATE_STOPPED);
+        if (top->state != UA_LIFECYCLESTATE_STOPPED) {
+            UA_LOG_ERROR(server->config.logging, UA_LOGCATEGORY_SERVER,
+                "Assertion `top->state == UA_LIFECYCLESTATE_STOPPED' failed.");
+        }
         top->clear(top);
         ZIP_REMOVE(UA_ServerComponentTree, &server->serverComponents, top);
         UA_free(top);
@@ -1350,6 +1354,8 @@ UA_StatusCode
 UA_Server_run_shutdown(UA_Server *server) {
     if(server == NULL)
         return UA_STATUSCODE_BADINVALIDARGUMENT;
+    UA_LOG_DEBUG(server->config.logging, UA_LOGCATEGORY_SERVER,
+        "The server is shutting down...");
 
     lockServer(server);
 
@@ -1362,6 +1368,8 @@ UA_Server_run_shutdown(UA_Server *server) {
 
     /* Set to stopping and notify the application */
     setServerLifecycleState(server, UA_LIFECYCLESTATE_STOPPING);
+    UA_LOG_DEBUG(server->config.logging, UA_LOGCATEGORY_SERVER,
+        "The server is set to STOPPING");
 
 #if UA_MULTITHREADING >= 100
     /* Stop regular callback for async operation processing */
@@ -1373,14 +1381,21 @@ UA_Server_run_shutdown(UA_Server *server) {
         removeCallback(server, server->houseKeepingCallbackId);
         server->houseKeepingCallbackId = 0;
     }
+    UA_LOG_DEBUG(server->config.logging, UA_LOGCATEGORY_NETWORK,
+        "The server housekeeping tasks: remove callbacks done");
 
     /* Stop all ServerComponents */
+    UA_LOG_DEBUG(server->config.logging, UA_LOGCATEGORY_SERVER,
+        "Stop server components...");
     ZIP_ITER(UA_ServerComponentTree, &server->serverComponents,
              stopServerComponent, NULL);
 
     /* Are we already stopped? */
     if(testStoppedCondition(server)) {
         setServerLifecycleState(server, UA_LIFECYCLESTATE_STOPPED);
+        UA_LOG_DEBUG(server->config.logging, UA_LOGCATEGORY_SERVER,
+            "The server is already STOPPED");
+
     }
 
     /* Only stop the EventLoop if it is coupled to the server lifecycle  */
@@ -1392,18 +1407,30 @@ UA_Server_run_shutdown(UA_Server *server) {
     /* Unlock and do one "normal" iteration. This allows threads waiting for the
      * server lock to proceed before the server lock is destroyed. */
     unlockServer(server);
+    UA_LOG_DEBUG(server->config.logging, UA_LOGCATEGORY_SERVER,
+        "Server: A last normal iteration");
     UA_Server_run_iterate(server, true);
     lockServer(server);
 
     /* Iterate the EventLoop until the server is stopped */
+    UA_LOG_DEBUG(server->config.logging, UA_LOGCATEGORY_SERVER,
+        "Iterate the EventLoop until the server is stopped");
     UA_StatusCode res = UA_STATUSCODE_GOOD;
     UA_EventLoop *el = server->config.eventLoop;
+    int i = 0;
     while(!testStoppedCondition(server) &&
-          res == UA_STATUSCODE_GOOD) {
+          res == UA_STATUSCODE_GOOD && i < 10) {
         res = el->run(el, 100);
+        i++;
+    }
+    if (i == 10) {
+        UA_LOG_ERROR(server->config.logging, UA_LOGCATEGORY_SERVER,
+            "Server components are not all stopped after 10 iterations");
     }
 
     /* Stop the EventLoop. Iterate until stopped. */
+    UA_LOG_DEBUG(server->config.logging, UA_LOGCATEGORY_SERVER,
+        "Stopping the EventLoop: iterate until stopped.");
     el->stop(el);
     while(el->state != UA_EVENTLOOPSTATE_STOPPED &&
           el->state != UA_EVENTLOOPSTATE_FRESH &&
-- 
2.51.0

