From 2acb3a009eedf1697e4d724f4089e6f70190bfe9 Mon Sep 17 00:00:00 2001
From: vdh-anssi <vincent.diemunsch@inria.fr>
Date: Tue, 20 Jan 2026 10:09:19 +0100
Subject: [PATCH 3/3] Added a memory leak to test ASAN

---
 src/server/ua_server_binary.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/server/ua_server_binary.c b/src/server/ua_server_binary.c
index 81363e7f1..e02d9a11d 100644
--- a/src/server/ua_server_binary.c
+++ b/src/server/ua_server_binary.c
@@ -217,6 +217,9 @@ processHEL(UA_Server *server, UA_SecureChannel *channel, const UA_ByteString *ms
                channel->state != UA_SECURECHANNELSTATE_RHE_SENT))
         return UA_STATUSCODE_BADINTERNALERROR;
 
+    UA_String leak = UA_STRING_ALLOC("Adding a memory leak\0");
+    UA_LOG_WARNING(server->config.logging,  UA_LOGCATEGORY_SERVER, (const char*) leak.data);
+
     size_t offset = 0; /* Go to the beginning of the TcpHelloMessage */
     UA_TcpHelloMessage helloMessage;
     UA_StatusCode retval =
-- 
2.51.0

