diff --git a/include/openssl/ssl.h b/include/openssl/ssl.h
--- a/include/openssl/ssl.h
+++ b/include/openssl/ssl.h
@@ -175,2 +175,4 @@
 OPENSSL_EXPORT int SSL_get_rfd(const SSL *ssl);
+
+OPENSSL_EXPORT int PUFFIN_extract_transcript(SSL* ssl, uint8_t* out, size_t* out_len);
 
diff --git a/ssl/ssl_lib.cc b/ssl/ssl_lib.cc
--- a/ssl/ssl_lib.cc
+++ b/ssl/ssl_lib.cc
@@ -359,3 +359,15 @@
 BSSL_NAMESPACE_END
+
+int PUFFIN_extract_transcript(SSL *ssl, uint8_t *out, size_t *out_len) {
+  bssl::SSL_HANDSHAKE *hs = ssl->s3->hs.get();
+
+  if (hs == NULL) {
+    return 0;
+  } else {
+    hs->transcript.GetHash(out, out_len);
+  }
+
+  return 1;
+}
 
 using namespace bssl;

