diff --git a/crypto/rand/deterministic.cc b/crypto/rand/deterministic.cc
--- a/crypto/rand/deterministic.cc
+++ b/crypto/rand/deterministic.cc
@@ -36,16 +36,12 @@
-static uint64_t g_num_calls = 0;
-static StaticMutex g_num_calls_lock;
-
-void RAND_reset_for_fuzzing() { g_num_calls = 0; }
+static thread_local uint64_t g_num_calls = 0;
+void RAND_reset_for_fuzzing() { g_num_calls = 0; }
 
 void bssl::CRYPTO_init_sysrand() {}
 
 void bssl::CRYPTO_sysrand(uint8_t *out, size_t requested) {
   static const uint8_t kZeroKey[32] = {0};
 
-  g_num_calls_lock.LockWrite();
   uint64_t num_calls = g_num_calls++;
-  g_num_calls_lock.UnlockWrite();
 
   uint8_t nonce[12];
   OPENSSL_memset(nonce, 0, sizeof(nonce));
